stubs/seaborn/seaborn/axisgrid.pyi

```diff
@@ -0,0 +1,243 @@
+        col_order: Iterable[Any] | None = None,
+        hue_order: Iterable[Any] | None = None,
+        hue_kws: dict[str, Any] | None = None,
+        dropna: bool = False,
+        legend_out: bool = True,
+        despine: bool = True,
+        margin_titles: bool = False,
+        xlim: tuple[float, float] | None = None,
+        ylim: tuple[float, float] | None = None,
+        subplot_kws: dict[str, Any] | None = None,
+        gridspec_kws: dict[str, Any] | None = None,
+    ) -> None: ...
+    def facet_data(self) -> Generator[tuple[tuple[int, int, int], DataFrame], None, None]: ...
+    def map(self, func: Callable[..., object], *args: str, **kwargs: Any) -> Self: ...
+    def map_dataframe(self, func: Callable[..., object], *args: str, **kwargs: Any) -> Self: ...
+    def facet_axis(self, row_i: int, col_j: int, modify_state: bool = True) -> Axes: ...
+    def despine(
+        self,
+        *,
+        fig: Figure | None = None,
+        ax: Axes | None = None,
+        top: bool = True,
+        right: bool = True,
+        left: bool = False,
+        bottom: bool = False,
+        offset: int | Mapping[str, int] | None = None,
+        trim: bool = False,
+    ) -> Self: ...
+    def set_axis_labels(
+        self, x_var: str | None = None, y_var: str | None = None, clear_inner: bool = True, **kwargs: Any
+    ) -> Self: ...
+    def set_xlabels(self, label: str | None = None, clear_inner: bool = True, **kwargs: Any) -> Self: ...
+    def set_ylabels(self, label: str | None = None, clear_inner: bool = True, **kwargs: Any) -> Self: ...
+    def set_xticklabels(self, labels: Iterable[str | Text] | None = None, step: int | None = None, **kwargs: Any) -> Self: ...
+    def set_yticklabels(self, labels: Iterable[str | Text] | None = None, **kwargs: Any) -> Self: ...
+    def set_titles(
+        self, template: str | None = None, row_template: str | None = None, col_template: str | None = None, **kwargs: Any
+    ) -> Self: ...
+    def refline(
+        self, *, x: float | None = None, y: float | None = None, color: ColorType = ".5", linestyle: str = "--", **line_kws: Any
+    ) -> Self: ...
+    @property
+    def axes(self) -> NDArray[Incomplete]: ...  # array of `Axes`
+    @property
+    def ax(self) -> Axes: ...
+    @property
+    def axes_dict(self) -> dict[Any, Axes]: ...
+
+class PairGrid(Grid):
+    x_vars: list[str]
+    y_vars: list[str]
+    square_grid: bool
+    axes: NDArray[Incomplete]  # two-dimensional array of `Axes`
+    data: DataFrame
+    diag_sharey: bool
+    diag_vars: NDArray[Incomplete] | None  # array of `str`
+    diag_axes: NDArray[Incomplete] | None  # array of `Axes`
+    hue_names: list[str]
+    hue_vals: Series[Incomplete]
+    hue_kws: dict[str, Any]
+    palette: _RGBColorPalette
+    def __init__(
+        self,
+        data: DataFrame,
+        *,
+        hue: str | None = None,
+        vars: Iterable[str] | None = None,
+        x_vars: Iterable[str] | None = None,
```

Comment:
These are documented as "lists of variable names" but I annotated them as iterables because they are casted to list at runtime. The `isscalar` check seems to allow `str` to be accepted but this is already covered by `Iterable[str]`. I think no change is needed here?

Reply:
Doesn't feel great to rely on the fact that `str` is an `Iterable[str]`, as usually that isn't what you want. As I recall pytype generally disallows str for Iterable[str]. So let's be explicit:
```suggestion
        x_vars: Iterable[str] | str | None = None,
```